README.md

```diff
@@ -20,11 +20,42 @@ Or install it yourself as:
 
     $ gem install ejson-rails
 
+## Configuration
+
+By default, the gem will look for decrypted secrets in `project/config/secrets.json` or `project/config/secrets.{current_rails_environment}.json` if that doesn't exist.
+
+If your application or environment has a unique way of retrieving decrypted secrets, you can do so by setting `EJSON::Rails::Railtie.ejson_secret_source` to a callable object in `config/application.rb`. For example:
+
+```ruby
+# config/application.rb
+
+# This must be placed BEFORE your application constant which inherits from Rails::Application
+EJSON::Rails::Railtie.ejson_secret_source = FooBar::SecretCredentialReader
+
+# Custom credential reader that lives somewhere else
+module FooBar
+  class SecretCredentialReader
+    class << self
+      def call
+        '{"secret": "secret_from_ejson_secret_source"}'
+      end
+    end
+  end
+end
+```
+
+For simple cases, you can use a `proc`:
+
+```ruby
+EJSON::Rails::Railtie.ejson_secret_source = proc { '{"secret": "secret_from_ejson_secret_source"}' }
```

Comment:
This kind of configs should not be in the Railtie. Let's move to the application like all other configs. `set_secrets` is an internal config that should go aways soon.

Reply:
I'd like to move these out of the railtie if we can find a better alternative, but I think moving it to application config would lead to a race condition.

The functionality in this gem lives in a `before_configuration` hook because we need to populate credentials as soon as the constant inherits from `Rails::Application`. That needs to happen to ensure the values are available for referencing when setting other application config values. For example:

```ruby
module FooBarApplication
  class Application < Rails::Application
    api_key = Rails.application.credentials.foo_bar_api_key&.fetch(:api_key)
  end
end
```

If we moved these into application config, our `before_configuration` hook will execute before our application configs are set.

Reply:
The `before_configuration` hook actually executes before the body of the application. This happens by design because this hook allows you to set any context needed before configuration happens. It is implemented as an explicit `ActiveSupport.run_load_hooks(:before_configuration, base)` call in the [base Rails::Application.inherited definition](https://github.com/rails/rails/blob/db30dd6fe71341ac132bb1e850bd4df2596a2803/railties/lib/rails/application.rb#L77).

Here is an example in action - we can see that the secret source was set outside the application body, and the `before_configuration` block executed before the body of the application; populating the `Rails.application.secrets` so the value can be referenced inside the body. 

```ruby
EJSON::Rails::Railtie.ejson_secret_source = proc { '{"hello_there": "I am a secret value!"}' }

module SomethingApp
  class Application < Rails::Application
    config.load_defaults(7.1)
    config.action_dispatch.default_headers = { "TEST" => Rails.application.secrets.hello_there }
  end
end
```

```bash
$  bundle exec rails console

irb(main):001> Rails.application.config.action_dispatch.default_headers
=> {"TEST"=>"I am a secret value!"}
```

If you'd like, I can point you at some production examples where an application body is referencing secrets which were first loaded by the ejson-rails`before_configuration` block.